Source/Concurrency/CivlTypeChecker.cs

```diff
@@ -144,21 +144,29 @@ public void TypeCheck()
       TypeCheckGlobalVariables();
       TypeCheckLemmaProcedures();
       TypeCheckYieldInvariants();
-
-      TypeCheckActionDecls();
+      
+      TypeCheckActions();
```

Comment:
Probably worth adding a comment about the two-step process of checking/initializing actions and the dependencies between the type checking phases. For example, I'm not sure why the "finish initialization" work cannot be done at the end of `TypeCheckActions`. I guess previously we tried to do all declaration checking first, and then do implementation checking. But now `TypeCheckActions` anyway already looks at action implementations.